src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Platform.java

```diff
@@ -24,12 +24,30 @@
  */
 package jdk.tools.jlink.internal;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteOrder;
+import java.util.HashMap;
 import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
 
 /**
  * Supported platforms
  */
-public record Platform(OperatingSystem os, Architecture arch) {
+public record Platform(OperatingSystem os, Architecture arch, ByteOrder endianness) {
+    private static final Properties KNOWN_ENDIANNESS;
+
+    static {
+        Properties p = null;
+        try (InputStream is = Platform.class.getResourceAsStream("target.properties")) {
+            p = new Properties();
+            p.load(is);
+        } catch (IOException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+        KNOWN_ENDIANNESS = p;
```

Comment:
The naming/usage suggests the value is solely for endianness but I think we want target properties to be able to support other attributes of the target too. This means "endianness" in the property name or the property value is something like a comma separate list of attributes, I don't have a strong opinion on this. But maybe your proposal is just to get the target.properties in place and we change it later?

Reply:
I read this as for endianness only and so the filename should be "target-endianness.properties" or something like that.   OTOH,  I like keeping it `target.properties` and add `.endianness` as the property name suffix will make it extensible to support other attributes easily.

Reply:
Given these inputs, I've now updated the PR to use `.endianness` as a suffix to individual keys in the properties file, to allow us reuse this `target.properties` for other details.

Existing tests continue to pass and manual testing shows the correct endianness is detected during cross platform image creation.

I'll trigger a tier testing to make sure nothing breaks.